### DISCUSSION TITLE
2-column hiccup with accept_page_break()

### DISCUSSION THREAD

**Discussion by resnbl** [py-pdf/fpdf2#314] [General]
I had some code which performed well for single column output, but when I implemented `accept_page_break()` for 2-column output it broke in unexpected ways. After some debugging, I traced the cause to the fact that this method may be called during `multi_cell(split_only=True)`. Since `accept_page_break()` changes state (col# and y position), and I was not expecting any changes when "just splitting a string", the my code following the split broke the output.

I don't really expect any changes to the package code, but I do think it may be prudent to call out the fact that `multi_cell()`, as well as `will_page_break()` (although this one didn't seem to bother me) will also invoke `accept_page_break()`.

Other than that, I am enjoying using this for some personal projects.

**Comment by Lucas-C** [comment]
Thanks for the feedback @resnbl !

To be sure to fully understand, the code that changed FPDF state _(col# and y position)_ was in your custom method overriding `accept_page_break()`?

**Reply by resnbl** [reply]
Yes, and the left margin was changed as well. (I started with Tutorial 4's implementation of `accept_page_break()` and `set_col()`...).
Update: left margin may be restored after `multi_cell()`, but my `accept_page_break()` didn't know that.

**Reply by Lucas-C** [reply]
I opened https://github.com/PyFPDF/fpdf2/pull/315 to improve things
Feel free to add review comments on it 😉

**Reply by Lucas-C** [reply]
PR has been merged & released: https://github.com/PyFPDF/fpdf2/blob/master/CHANGELOG.md#250---2022-01-22
